src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+        inputs_embeds = torch.zeros_like(queries_embeddings, requires_grad=True)
+
+        batch_size, num_channels, height, width = image_features.shape
+        # rearrange both image_features and position_embeddings "b c h w -> b (h w) c"
+        image_features = image_features.view(batch_size, num_channels, height * width).permute(0, 2, 1)
+        position_embeddings = position_embeddings.view(batch_size, num_channels, height * width).permute(0, 2, 1)
+
+        transformer_decoder_output: DetrDecoderOutput = self.transformer_decoder(
+            inputs_embeds=inputs_embeds,
+            attention_mask=None,
+            encoder_hidden_states=image_features,
+            encoder_attention_mask=None,
+            position_embeddings=position_embeddings,
+            query_position_embeddings=queries_embeddings,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=None,
+        )
+        return transformer_decoder_output
+
+
+MASKFORMER_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) sub-class. Use
+    it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`MaskFormerConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+MASKFORMER_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoFeatureExtractor`]. See
+            [`AutoFeatureExtractor.__call__`] for details.
+
+        pixel_mask (`torch.LongTensor` of shape `(batch_size, height, width)`, *optional*):
+            Mask to avoid performing attention on padding pixel values. Mask values selected in `[0, 1]`:
+
+            - 1 for pixels that are real (i.e. **not masked**),
+            - 0 for pixels that are padding (i.e. **masked**).
+
+            [What are attention masks?](../glossary#attention-mask)
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of Detr's decoder attention layers.
+
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~MaskFormerModelOutput`] instead of a plain tuple.
+"""
+
+
+class MaskFormerPretrainedModel(PreTrainedModel):
```

Comment:
```suggestion
class MaskFormerPreTrainedModel(PreTrainedModel):
```
The class should be named with a T

Reply:
changed it everywhere